src/libraries/System.IO.FileSystem/tests/Directory/CreateDirectory_UnixFileMode.Unix.cs

```diff
@@ -39,6 +39,21 @@ public void CreateDoesntChangeExistingMode()
             Assert.Equal(initialMode, sameDir.UnixFileMode);
         }
 
+        [Fact]
+        public void MissingParentsHaveDefaultPermissions()
+        {
+            string parent = GetRandomDirPath();
+            string child = Path.Combine(parent, "child");
+
+            const UnixFileMode childMode = UnixFileMode.UserRead | UnixFileMode.UserExecute;
+            DirectoryInfo childDir = Directory.CreateDirectory(child, childMode);
+
+            Assert.Equal(childMode, childDir.UnixFileMode);
+
+            UnixFileMode defaultPermissions = Directory.CreateDirectory(GetRandomDirPath()).UnixFileMode;
+            Assert.Equal(defaultPermissions, File.GetUnixFileMode(parent));
+        }
```

Comment:
Should there be a similar test for Tar?

Reply:
For `Tar` no directories get created as missing parents. Each directory gets created with an explicit mode.

Reply:
This is a good question, because we _do_ create missing parents. Let's assume someone creates an archive with two entries:

```
1) Directory:    parent/child/
2) RegularFile: parent/child/file.txt
```

Notice that there is no entry for `parent/`.

When we iterate through the entries, the first entry will be created with `Directory.CreateDirectory` and will automatically create the parent, which did not have an entry. This happens in `TarHelpers.Unix.cs`, line 94 in this PR.

Reply:
Actually, this scenario is already covered with existing unit tests, which were added in a previous PR:

`src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.File.Tests.cs` lines 184 and 189: Two directory entries are manually added to the archive without adding their dir parent before them.